### DISCUSSION TITLE
How to associate a term from one category to the terms from another category in a framework

### DISCUSSION THREAD

**Discussion by devendra-tarento** [Sunbird-Knowlg/Community#84] [Q&A] [answered]
I am trying to setup a framework which has following categories (along with terms) mapped to sunbird master categories:
1- Role (Board)
    R1
    R2
    ..
    ..
    ..
    R18

2- Activity (Medium)
     A1
     A2
     ...
     ...
     A136
    
3- Competency (Grade Level)
     C1
     C2
     ...
     ...
    C72

4- Competency Level (Subject)
     CL1
     CL2
      ...
      ...
     CL360
 
I have created above categories and their terms under a new framework using CREAT CATEGORY and CREAT TERM api as per the Framework creation document [Framework core apis](http://docs.sunbird.org/latest/apis/framework)

Now I want know how to achieve below requirement using TERM APIs:
1-  Select R1 from Role (Board)
2- Only show options (A1,  A2, .. ,.. ,.. , A13) to select from Activity (Medium)
3- When A1 selected from Activity (Medium)
4- Only show options (C1,  C2..,.., C5) to select from Competency (Grade Level)
5- When C1 is selected from  Competency (Grade Level)
6-  Only show options (CL1,  CL2) to select from Competency Level (Subject)

In the document it is mentioned to use UPDATE TERM api to associate terms from other categories like:

**End point URL** -> 
   api/framework/v1/term/update/R1?framework=upsmf&category=board
**Request Payload**
```
     {
      "request": {
          "term": {
              "associationswith": [
                  {
                      "identifier": ""
                  },
                  {
                      "identifier": ""
                  },
                  {
                      "identifier": ""
                  }
              ]
          }
      }
  }
```

Here I am not having clarity that which all identifiers should be passed if I want to achieve above mentioned requirement ?
1- Update R1 by passing A1 to A13 identifiers
2- Then Update A1 by passing C1 to C5 identifiers
3- And then update C1 by passing CL1 and CL2 identifiers 

Repeat this step for each Role (R1...R18), each Activity(A1, A2....,A136) and each Competency (C1,C2...,C72). 

Is my understanding correct or not ?

Please let me know what is the right way to do it in case if my understanding is wrong.

@vinukumar-vs

**Comment by Jayaprakash8887** [comment, marked as answer]
For associating a term with previous category terms, you will have to specify identifiers of the previous category terms.

CL2 should have associationswith identifiers of C1, A1, R1. Then CL2 is filtered only when R1, A1 and C1 are selected. 

https://knowlg.sunbird.org/learn/product-and-developer-guide/taxonomy-and-tagging/framework-service/faqs


Lets say CL2 is associated with only C1. Then CL2 will be visible for all associations C1 has. Meaning, If C1 has associations with A2 and R2; then CL2 will get filtered for R2-A2-C1 as well. If CL2 is associated with R1-A1-C1, then it will be filtered only for that combination.
